runtime/core/exec_aten/util/scalar_type_util.h

```diff
@@ -910,20 +910,21 @@ struct promote_types {
   }
 #endif
 
-#define ET_INTERNAL_SWITCH(TYPE, CONTEXT, NAME, ...) \
-  [&] {                                              \
-    const auto& _st = TYPE;                          \
-    constexpr const char* et_switch_name = NAME;     \
-    (void)et_switch_name; /* Suppress unused var */  \
-    switch (_st) {                                   \
-      __VA_ARGS__                                    \
-      default:                                       \
-        ET_CHECK_MSG(                                \
-            false,                                   \
-            "Unhandled dtype %s for %s",             \
-            ::executorch::runtime::toString(_st),    \
-            et_switch_name);                         \
-    }                                                \
+#define ET_INTERNAL_SWITCH(TYPE, CONTEXT, NAME, ...)           \
+  [&] {                                                        \
+    const auto& _st = TYPE;                                    \
+    constexpr const char* et_switch_name = NAME;               \
+    (void)et_switch_name; /* Suppress unused var */            \
+    switch (_st) {                                             \
+      __VA_ARGS__                                              \
+      default:                                                 \
+        CONTEXT.fail(torch::executor::Error::InvalidArgument); \
```

Comment:
Nit: Leave a comment along the lines "this sets the return code which should be checked by the caller"